src/Http/Http.Extensions/gen/StaticRouteHandlerModel/Emitters/DiagnosticEmitter.cs

```diff
@@ -0,0 +1,44 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+
+namespace Microsoft.AspNetCore.Http.RequestDelegateGenerator.StaticRouteHandlerModel;
+
+internal static class DiagnosticEmitter
+{
+    public static void EmitRequiredDiagnostics(this EndpointResponse response, List<Diagnostic> diagnostics, Location location)
+    {
+        if (response.ResponseType is ITypeParameterSymbol)
+        {
+            diagnostics.Add(Diagnostic.Create(DiagnosticDescriptors.TypeParametersNotSupported, location));
+        }
+        if (response.ResponseType?.DeclaredAccessibility is Accessibility.Private or Accessibility.Protected)
```

Comment:
> Unless these are meant to be else if

I debated this but I've always like the upfront disclosure you get when all the warnings are emitted at once instead of the progressive disclosure that allows you to keep fixing things until you get it right.

Reply:
Have you used this in a generator? It seems like the API works by assessing accessibility within the context of an _existing_ syntax tree, but here we're trying to check accessibility in code we haven't generated yet.